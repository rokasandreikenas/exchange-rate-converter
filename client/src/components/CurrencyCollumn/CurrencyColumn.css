.currency-column-container {
  width: 40%;
  padding: 30px 10px;
}

@media (max-width: 768px) {
  .currency-column-container {
    width: 90%;
  }
}

.currency-column-title {
  font-size: 11px;
  margin-bottom: 5px;
  color: rgba(0, 0, 0, 0.6);
  text-transform: uppercase;
}

.currency-column-input {
  position: relative;
  margin-top: 20px;
}

.currency-column-input span.currency {
  font-size: 22px;
  font-weight: 100;
  position: absolute;
  color: rgba(0, 0, 0, 0.6);
  margin-top: 33px;
}

.currency-column-input input {
  height: 100px;
  width: 100%;
  border: none;
  outline: none;
  font-size: 52px;
  font-weight: 200;
  text-indent: 50px;
  margin-bottom: 20px;
  color: rgba(0, 0, 0, 0.6);
  background: transparent;
  border-bottom: 1px solid #eee;
}
/*# sourceMappingURL=CurrencyColumn.css.map */