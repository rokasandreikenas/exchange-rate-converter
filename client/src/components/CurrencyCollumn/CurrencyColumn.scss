.currency-column {
  &-container {
    width: 40%;
    padding: 15px;
  }

  &-title {
    font-size: 11px;
    margin-bottom: 5px;
    color: rgba(0, 0, 0, 0.6);
    text-transform: uppercase;
  }

  &-input {
    position: relative;
    margin-top: 20px;

    span.currency {
      // bottom: 40px;
      font-size: 22px;
      font-weight: 100;
      position: absolute;
      color: rgba(0, 0, 0, 0.6);
      margin-top: 33px;
    }
  }

  &-input input {
    height: 100px;
    width: 100%;
    border: none;
    outline: none;
    font-size: 52px;
    font-weight: 200;
    text-indent: 50px;
    margin-bottom: 20px;
    color: rgba(0, 0, 0, 0.6);
    background: transparent;
    border-bottom: 1px solid #eee;
  }
}
