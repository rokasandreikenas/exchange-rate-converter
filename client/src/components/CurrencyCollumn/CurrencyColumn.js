import React from "react";
import PropTypes from "prop-types";
import Select from "react-select";
import getSymbolFromCurrency from "currency-symbol-map";
import "./CurrencyColumn.scss";

const CurrencyColumn = ({
  title,
  selectedOption,
  setSelectedOption,
  options,
  inputValue,
  setInputValue,
  defaultValue,
  readOnly,
}) => {
  console.log(selectedOption, inputValue, defaultValue);
  return (
    <div className="currency-column-container">
      <div className="currency-column-title">{title}</div>
      <Select
        value={selectedOption}
        onChange={(e) => setSelectedOption(e)}
        options={options}
        placeholder="Choose currency..."
        // defaultValue={defaultValue}
      />
      <div className="currency-column-input">
        <span className="currency">
          {getSymbolFromCurrency(defaultValue.label)}
        </span>
        <input
          name="amount"
          placeholder="0"
          value={inputValue}
          onChange={(e) => setInputValue(e.currentTarget.value)}
          // defaultValue={defaultValue.value}
          readOnly={readOnly}
        />
      </div>
    </div>
  );
};

export default CurrencyColumn;

CurrencyColumn.propTypes = {
  title: PropTypes.string.isRequired,
  selectedOption: PropTypes.any,
  setSelectedOption: PropTypes.func.isRequired,
  options: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  inputValue: PropTypes.number,
  setInputValue: PropTypes.func,
};
