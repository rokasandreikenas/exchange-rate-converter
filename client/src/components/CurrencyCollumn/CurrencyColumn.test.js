import React from "react";
import CurrencyColumn from "./CurrencyColumn";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("CurrencyColumn", () => {
  it("renders CurrencyColumn component correctly ", () => {
    const selectedOptionMock = { value: 1, label: "mock" };
    const optionsMock = [
      { value: 1, label: "mock1", value: 2, label: "mock2" },
    ];

    const output = shallow(
      <CurrencyColumn
        title="Mock title"
        selectedOption={selectedOptionMock}
        setSelectedOption={() => {}}
        options={optionsMock}
        inputValue={selectedOptionMock.value}
        setInputValue={() => {}}
        readOnly={false}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
