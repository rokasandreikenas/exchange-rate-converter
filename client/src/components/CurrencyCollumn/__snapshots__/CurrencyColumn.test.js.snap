// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyColumn renders CurrencyColumn component correctly  1`] = `
<div
  className="currency-column-container"
>
  <div
    className="currency-column-title"
  >
    Mock title
  </div>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "mock2",
          "value": 2,
        },
      ]
    }
    placeholder="Choose currency..."
    value={
      Object {
        "label": "mock",
        "value": 1,
      }
    }
  />
  <div
    className="currency-column-input"
  >
    <span
      className="currency"
    />
    <input
      name="amount"
      onChange={[Function]}
      placeholder="0"
      readOnly={false}
      type="number"
      value={1}
    />
  </div>
</div>
`;
